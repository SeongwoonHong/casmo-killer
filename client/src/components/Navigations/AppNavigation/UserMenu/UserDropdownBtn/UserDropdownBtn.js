import React from 'react';
import PropTypes from 'prop-types';

import TopNavBtn from '../../../shared/TopNavBtn';

import './UserDropdownBtn.scss';

const UserDropdownBtn = ({
  isLoggedIn, displayName, avatar, level, toggleUserMenu
}) => {

  return (
    <TopNavBtn
      className={
        isLoggedIn
        ? 'User-dropdown-btn__btn--loggedIn'
        : ''
      }
      classPrefix="User-dropdown-btn"
      icon="person_outline"
      onClick={ toggleUserMenu }>
      {
        isLoggedIn && displayName
          ? (
            <div className="User-dropdown-btn__btn__row">
              <div className="User-dropdown-btn__btn__avatar">
                {
                  avatar
                  ? (
                    <img
                      className="User-dropdown-btn__btn__avatar__img"
                      src={ avatar }
                      alt={ displayName } />
                  )
                  : (
                    <span className="User-dropdown-btn__btn__avatar__text">
                      { displayName[0] }
                    </span>
                  )
                }
              </div>
              <div className="User-dropdown-btn__btn__username">
                <span className="User-dropdown-btn__btn__username__text">
                  { displayName }
                </span>
                <span className="User-dropdown-btn__btn__username__level">
                  { level || 'newbie' }
                </span>
              </div>
              <i className="User-dropdown-btn__btn__icon material-icons">
                arrow_drop_down
              </i>
            </div>
        )
        : null
      }
    </TopNavBtn>
  );

};

UserDropdownBtn.propTypes = {
  isLoggedIn: PropTypes.bool,
  displayName: PropTypes.string,
  avatar: PropTypes.string,
  level: PropTypes.string,
  toggleUserMenu: PropTypes.func
};

UserDropdownBtn.defaultProps = {
  isLoggedIn: false,
  displayName: '',
  avatar: '',
  level: '',
  toggleUserMenu: () => {}
};

export default UserDropdownBtn;
