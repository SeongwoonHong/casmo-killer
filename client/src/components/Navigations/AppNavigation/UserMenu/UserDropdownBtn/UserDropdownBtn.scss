.User-dropdown-btn__btn {
    display: none;
    @media screen and (min-width: 992px) {
        @include flex-box();
        display: block;
        margin-left: auto;
        margin-right: 0;
        &--loggedIn {
            width: auto;
        }
        &__row {
            @include flex-box(space-between);
            height: $top-nav-size;
            width: $user-dropdown-width;
        }
        &__username {
            color: #fff;
            margin-left: 0;
            margin-right: auto;
            text-align: left;
            &__text,
            &__level {
                display: block;
                line-height: normal;
            }
            &__text {
                @include truncate();
                font-size: 12px;
            }
            &__level {
                color: $accent-colour;
                font-size: 11px;
                text-transform: capitalize;
            }
        }
        &__icon {
            @include rectanglify($top-nav-size);
            line-height: $top-nav-size;
        }
        &__avatar {
            margin: 0 10px;
            &__img,
            &__text {
                @include rectanglify($top-nav-size - 10px);
                border-radius: 50%;
                display: block;
            }
            &__text {
                background-color: palette(Teal, 100);
                font-size: 28px;
                line-height: 32px;
                text-align: center;
            }
        }
        &:hover {
            background-color: palette(Grey, 700);
            height: 50px;
        }
    }
}
