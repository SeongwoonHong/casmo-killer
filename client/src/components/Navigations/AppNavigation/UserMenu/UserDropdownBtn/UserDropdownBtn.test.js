/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';

import UserDropdownBtn from './index';

describe('<UserDropdownBtn />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <UserDropdownBtn />
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

});
