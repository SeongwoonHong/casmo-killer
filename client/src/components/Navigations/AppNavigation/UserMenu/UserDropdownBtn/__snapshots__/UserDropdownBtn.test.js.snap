// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDropdownBtn /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdownBtn
    avatar=""
    displayName=""
    isLoggedIn={false}
    level=""
    toggleUserMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": null,
      "className": "",
      "classPrefix": "User-dropdown-btn",
      "icon": "person_outline",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": null,
        "className": "",
        "classPrefix": "User-dropdown-btn",
        "icon": "person_outline",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
